api/src/main/java/org/openmrs/api/db/hibernate/HibernateAdministrationDAO.java

```diff
@@ -200,9 +202,51 @@ public void setApplicationContext(ApplicationContext applicationContext) throws
 	
 	/**
 	 * @see org.openmrs.api.db.AdministrationDAO#validate(java.lang.Object, Errors)
+	 * @should pass validation if all required fields have proper values
 	 */
 	@Override
 	public void validate(Object object, Errors errors) throws DAOException {
+		String name = object.getClass().getSimpleName();
+		Class entityClass = object.getClass();
+		for (Field field : object.getClass().getDeclaredFields()) {
+			field.setAccessible(true);
+			if (field.getName().equalsIgnoreCase(name)) {
+				Object value = null;
+				try {
+					value = field.get(object);
+				} catch (IllegalArgumentException e) {
+					log.warn("Unable to get the value of " + field.getName());
+				} catch (IllegalAccessException e) {
+					log.warn("Unable to get the value of " + field.getName());
+				}
+				if (value != null) {
+					int l1 = getMaximumPropertyLength(entityClass, field.getName());
+					int l2 = value.toString().length();
+					if (l2 > l1) {
+						errors.rejectValue(field.getName(),  "error.exceededMaxLengthOfField",new Object[] { l1 }, null);
+					}
+				}
+			}
+		}
+
+		ClassMetadata metadata = sessionFactory.getClassMetadata(entityClass);
+		if(metadata != null){
+			String[] propNames = metadata.getPropertyNames();
+			for (int i = 0; i < propNames.length; i++) {
+				String propType = metadata.getPropertyType(propNames[i]).getName();
+				if (propType.equalsIgnoreCase("String")) {
+					int p1 = getMaximumPropertyLength(entityClass, propNames[i]);
```

Comment:
I was on the 1.12.x branch, i see this was fixed in 2.0.x and master at 41207c6c8a065a5af97be14807a7fe2cb965fe0b. Previously the method actually used column name but it has been fixed ever since.